crates/orderbook/src/database/orders.rs

```diff
@@ -529,6 +610,37 @@ fn is_buy_order_filled(amount: &BigDecimal, executed_amount: &BigDecimal) -> boo
     !executed_amount.is_zero() && *amount == *executed_amount
 }
 
+#[derive(Serialize, PartialEq, Debug, Clone)]
```

Comment:
I'm pretty sure this is not the right module for this logic to live in (look at the size of the module). If you look at the other functions in this module they are mainly plain read/write functions. It looks like you are also defining the actual api types here (in the db model) for which I cannot see any precedence.

Can we create a separate module just for order status that defines all of this and contains the logic around fetching and converting the raw competition result into what the fronted expects please.